src/renderer/components/editor.tsx

```diff
@@ -113,6 +119,30 @@ export class Editor extends React.Component<EditorProps> {
     }
   }
 
+  /**
+   * Implements external url handling for Monaco.
+   *
+   * @returns {MonacoType.editor.IOpenerService}
+   */
+  private openerService() {
+    const { appState } = this.props;
+
+    return {
+      open: (url: string) => {
+        appState
+          .showConfirmDialog({
+            label: `Open ${url} in external browser?`,
+            ok: 'Open',
+          })
+          .then((open) => {
+            if (open) {
+              window.ElectronFiddle.openExternal(url);
+            }
+          });
+      },
+    };
+  }
+
```

Comment:
Can we move this to an utility function (maybe in `/src/utils/editor-utils.ts`) to avoid duplication?

Reply:
i'm open to this but it feels a little strange passing appState to a util function 🤔

Reply:
Fair point, we do access `appState` via `window.app.state` elsewhere though, this variable is so ubiquitous that I think it's acceptable — but either way is fine by me, I don't feel strongly.